pennylane/qchem/openfermion_obs.py

```diff
@@ -881,22 +857,59 @@ def molecular_hamiltonian(
             corresponding to the qubit number equal to its index.
             For type dict, only int-keyed dict (for qubit-to-wire conversion) is accepted for
             partial mapping. If None, will use identity map.
-        alpha (array[float]): exponents of the primitive Gaussian functions
-        coeff (array[float]): coefficients of the contracted Gaussian functions
         args (array[array[float]]): initial values of the differentiable parameters
         load_data (bool): flag to load data from the basis-set-exchange library
         convert_tol (float): Tolerance in `machine epsilon <https://numpy.org/doc/stable/reference/generated/numpy.real_if_close.html>`_
             for the imaginary part of the Hamiltonian coefficients created by openfermion.
             Coefficients with imaginary part less than 2.22e-16*tol are considered to be real.
 
+    .. warning::
+        Use of ``qml.qchem.molecular_hamiltonian`` with symbols and geometry arguments is deprecated. Instead, please use function with ``qml.Molecule`` object``.
+
+    .. details::
+        :title: Usage Details
+
+        Old interface used molecular_hamiltonian function with molecular information as separate arguments:
+        molecular_hamiltonian(symbols, coordinates, name="molecule", charge=0, mult=1, basis="sto-3g", method="dhf",
+        active_electrons=None, active_orbitals=None, mapping="jordan_wigner", outpath=".", wires=None, alpha=None,
+        coeff=None, args=None, load_data=False, convert_tol=1e012)
+
+        Additional Parameters:
+          symbols (list[str]): symbols of the atomic species in the molecule
+
+          coordinates (array[float]): atomic positions in Cartesian coordinates.
+          The atomic coordinates must be in atomic units and can be given as either a 1D array of
+          size ``3*N``, or a 2D array of shape ``(N, 3)`` where ``N`` is the number of atoms.
+          name (str): name of the molecule
+
+          charge (int): Net charge of the molecule. If not specified a neutral system is assumed.
+
+          mult (int): Spin multiplicity :math:`\\mathrm{mult}=N_\\mathrm{unpaired} + 1` for :math:`N_\\mathrm{unpaired}`
+          unpaired electrons occupying the HF orbitals. Possible values of ``mult`` are :math:`1, 2, 3, \\ldots`.
+          If not specified, a closed-shell HF state is assumed.
+
+          basis (str): atomic basis set used to represent the molecular orbitals
+
+          .. code-block:: python
+
+              from pennylane import qchem
+
+              symbols = ["H", "H"]
+              geometry = [[0.0, 0.0, 0.0], [0.0, 0.0, 1.0]]
+
+              H, qubit = qchem.molecular_hamiltonian(symbols, geometry, charge=0)
+
+        This information can now be wrapped in Molecule object and provided to the molecular_hamiltonian function in the new interface.
```

Comment:
```suggestion
        Old interface used ``molecular_hamiltonian`` function with molecular information as separate arguments:

        .. py:function:: molecular_hamiltonian(symbols, coordinates, name="molecule", charge=0, mult=1, basis="sto-3g", method="dhf", active_electrons=None, active_orbitals=None, mapping="jordan_wigner", outpath=".", wires=None, alpha=None, coeff=None, args=None, load_data=False, convert_tol=1e12)

            Keyword Arguments:
                - **symbols** (list[str]): symbols of the atomic species in the molecule
                - **coordinates** (array[float]): atomic positions in Cartesian coordinates and atomic units as a 1D array of size ``3*N``, or a 2D array of shape ``(N, 3)`` where ``N`` is the number of atoms.
                - **name** (str): name of the molecule
                - **charge** (int): Net charge of the molecule. If not specified a neutral system is assumed.
                - **mult** (int): Spin multiplicity :math:`\\mathrm{mult}=N_\\mathrm{unpaired} + 1` for :math:`N_\\mathrm{unpaired}` unpaired electrons occupying the HF orbitals. By default, a closed-shell HF state is assumed.
                - **basis** (str): atomic basis set used to represent the molecular orbitals

        This interface was supposed to be use as follows:

        .. code-block:: python

            from pennylane import qchem

            symbols = ["H", "H"]
            geometry = [[0.0, 0.0, 0.0], [0.0, 0.0, 1.0]]

            H, qubit = qchem.molecular_hamiltonian(symbols, geometry, charge=0)

        This information can now be wrapped in a :class:`~.qchem.molecule.Molecule` object and provided to the ``molecular_hamiltonian`` function in the new interface.
```

I think this helps with formatting. Feel free to use something similar.

Reply:
Also, I think `args` (for differentiable parameters - `alpha`, `coeffs`) have also changed. We should mention them as well?